akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -445,6 +445,27 @@ object Source {
       (s: S) ⇒ read.apply(s).toScala.map(_.asScala)(akka.dispatch.ExecutionContexts.sameThreadExecutionContext),
       (s: S) ⇒ close.apply(s).toScala))
 
+  /**
+   * Upcast a stream of elements to a stream of supertypes of that element. Useful in combination with
+   * fan-in combinators where you do not want to pay the cost of casting each element in a `map`.
+   *
+   * Example:
+   *
+   * {{{
+   * Source<Apple, NotUsed> apples = null;
+   * Source<Orange, NotUsed> oranges = null;
```

Comment:
perhaps use `Source.empty` in such examples instead of null

Reply:
Ups, yes, indeed, copy pasta from the test.